src/components/ToolbarComponent.js

```diff
@@ -51,6 +57,49 @@ class ToolbarComponent {
       this.#showSummaryKey
     );
     this.components.fullScreen.addEventListener("click", this.#showFullScreen);
+
+    this.components.questionMarkHighlight.addEventListener(
+      "mouseenter",
+      (event) => {
+        event.preventDefault();
+        EventHandlerService.publish(PDFLEvents.onShowHighlightTutorialWindow);
```

Comment:
Since the event bus can take one or more parameters after the event name, I suggest (but it can be left as it is) to use a unique event "onShowTutorialWindow" and pass a literal identifier of the tutorial type as a second parameter of the publish. That way is cleaner for two reasons: 1) the event is always the same and change only a property of it (like the shortcuts, it is always a keyup event), 2) The subscription of all tutorial event happens in the same class

Note: the hide event is correct to be separate

Reply:
I agree with you but not completely because that way you would still need something like a switch case with **hardcoded strings** like there was before which is better to avoid.